knode/validator/validator.go

```diff
@@ -191,8 +192,14 @@ func (val *validator) SetCoinbase(walletAccount accounts.WalletAccount) error {
 	return nil
 }
 
-func (val *validator) SetDeposit(deposit *big.Int) {
+func (val *validator) SetDeposit(deposit *big.Int) error {
+	if !val.Validating() {
```

Comment:
this is not true, you can only setDeposit if NOT running

Reply:
Old logic was `you can set deposit in any time`. For me, it's incorrect and leads to errors.

Reply:
this is confusing, can we remove all SetDeposit and just assume the Start NEEDs the deposit value